services/workers/src/test/java/org/sagebionetworks/table/worker/MaterializedViewUpdateWorkerIntegrationTest.java

```diff
@@ -355,6 +355,68 @@ public void testMaterializedViewWithLeftJoinViews() throws Exception {
 		}, MAX_WAIT_MS);
 	}
+
+	/*
+	 * This test is added to reproduce and fix PLFM-7321 user access issue.
+	 * */
+	@Test
+	public void testMaterializedViewWithLeftJoinViewsWithNormalUser() throws Exception {
+		int numberOfFiles = 5;
+		final List<Entity> leftEntities = createProjectHierachy(numberOfFiles);
+		final String leftFolderId = leftEntities.get(2).getId();
+		aclDaoHelper.update(leftFolderId, ObjectType.ENTITY, a -> {
+			a.getResourceAccess().add(createResourceAccess(userInfo.getId(), ACCESS_TYPE.READ));
+		});
+
+		final List<String> fileIds = leftEntities.stream()
+				.filter((e) -> e instanceof FileEntity)
+				.map(e -> e.getId())
+				.collect(Collectors.toList());
+		assertEquals(numberOfFiles, fileIds.size());
+
+		final List<PatientData> patientDataLeft = Arrays.asList(
+				new PatientData().withCode("abc").withPatientId(111L),
+				new PatientData().withCode("def").withPatientId(222L)
+		);
+
+		final List<Entity> rightEntities = createProjectHierachy(2);
+		final String rightFolderId = rightEntities.get(2).getId();
+		aclDaoHelper.update(rightFolderId, ObjectType.ENTITY, a -> {
+			a.getResourceAccess().add(createResourceAccess(userInfo.getId(), ACCESS_TYPE.READ));
+		});
+
+		final List<PatientData> patientDataRight = Arrays.asList(
+				new PatientData().withCode("abc").withPatientId(333L),
+				new PatientData().withCode("def").withPatientId(222L)
+		);
+
+		final IdAndVersion leftId = createFileViewWithPatientIds(leftEntities, patientDataLeft);
+
+		final IdAndVersion rightId = createFileViewWithPatientIds(rightEntities, patientDataRight);
+
+		final String definingSql = String.format(
+				"select l.id, l.patientId, r.patientId from %s l left join %s r on (l.patientId = r.patientId) order by l.ROW_ID",
+				leftId.toString(), rightId.toString());
+
+		final IdAndVersion materializedViewId = createMaterializedView(leftFolderId, definingSql);
+
+		final String materializedQuery = "select * from "+materializedViewId.toString()+" order by \"l.id\" asc";
+
+		final List<Row> expectedRows = Arrays.asList(
+				new Row().setRowId(1L).setVersionNumber(0L).setValues(Arrays.asList(fileIds.get(0), "111", null)),
+				new Row().setRowId(2L).setVersionNumber(0L).setValues(Arrays.asList(fileIds.get(1), "222", "222")),
+				new Row().setRowId(3L).setVersionNumber(0L).setValues(Arrays.asList(fileIds.get(2), "111", null)),
+				new Row().setRowId(4L).setVersionNumber(0L).setValues(Arrays.asList(fileIds.get(3), "222", "222")),
+				new Row().setRowId(5L).setVersionNumber(0L).setValues(Arrays.asList(fileIds.get(4), "111", null))
+		);
+
+
+		// Wait for the query against the materialized view to have the expected results.
+		asyncHelper.assertQueryResult(userInfo, materializedQuery, (results) -> {
+			System.out.println(" result size "+ results.getQueryResult().getQueryResults().getRows().size());
+			assertEquals(expectedRows, results.getQueryResult().getQueryResults().getRows());
+		}, MAX_WAIT_MS);
+	}
 	
```

Comment:
Lets add a test that covers the case where the user does not have permission to see some files on the left or the right.